fix_includes.py

```diff
@@ -1339,87 +1339,159 @@ def _GetFirstNamespaceLevelReorderSpan(file_lines):
+            namespace_prefixes.append('namespace %s { ' % namespace)
+          namespace_depth += 1
+        
+        namespace_reorder_spans[''.join(namespace_prefixes)] = (
+          line_number+1, line_number+1)
+
+      elif line_info.type == _NAMESPACE_START_ALLMAN_RE:
+        for namespace in _GetNamespaceNames(line_info.line):
+          if not namespace:
+            namespace_prefix += 'namespace'
+          else:
+            namespace_prefix += 'namespace %s' % namespace
+
+      elif line_info.type == _NAMESPACE_START_MIXED_RE:
+        # For mixed namespace styles, we need to append normalized prefixes
+        # using regular and Allman style.  Treat the first elements as
+        # normal and only treat the final element as Allman.  By the
+        # nature of mixed namespaces, there will always be more than
+        # one namespace so it is okay to assume that _GetNamespaceNames
+        # will always return multiple records.
+        namespaces = _GetNamespaceNames(line_info.line)
+        for namespace in namespaces[:-1]:
+          if not namespace:
+            namespace_prefixes.append('namespace { ')
+          else:
+            namespace_prefixes.append('namespace %s { ' % namespace)
+          namespace_depth += 1
+        else:
+          if not namespaces[-1]:
+            namespace_prefix += 'namespace'
+          else:
+            namespace_prefix += 'namespace %s' % namespaces[-1]
+
+      elif line_info.type == _NAMESPACE_CONTINUE_ALLMAN_MIXED_RE:
+        # Append to the simplified allman namespace
+        namespace_prefix += ' { '
+        namespace_prefixes.append(namespace_prefix)
+        namespace_depth += 1
+        namespace_reorder_spans[''.join(namespace_prefixes)] = (
+          line_number+1, line_number+1)
+        namespace_prefix = ''  # reset this in case we find more
+
+      elif line_info.type == _NAMESPACE_END_RE:
+        # remove C++ comments and count the ending brackets
+        namespace_end_count = line_info.line.split("/")[0].count("}")
+        for i in range(namespace_end_count):
+          namespace_depth -= 1
+          namespace_prefixes.pop()
+
+      elif line_info.type == _FORWARD_DECLARE_RE:
+        # If we're not in a namespace, keep going.  Otherwise, this is
+        # just the situation we're looking for!  Update the dictionary
+        # with the better reorder span
+        if namespace_depth > 0:
+          namespace_reorder_spans[''.join(namespace_prefixes)] = (
+            line_info.reorder_span)
+
+      elif line_info.type == None:
+        break
+
+      else:
+        # We should have handled all the cases above!
+        assert False, ('unknown line-info type',
+                       _LINE_TYPES.index(line_info.type))
+  except:
+    # namespace detection could be tricky so take what we have
+    pass
+
+  # return a reverse sorted list so longest matches are checked first
+  return sorted(list(namespace_reorder_spans.items()), reverse=True)
```

Comment:
I think Python 3 will return a generator from `sorted`, not a list. So depending on how this is used outside, you may need to materialize a list around `sorted`. In that vein, do you really need to materialize the `items()`? I would expect `sorted` to work on a generator/sequence-like thing.

Reply:
Sorry, no -- `sorted` can never return a generator, it (obviously) needs to materialize to sort. But the `list(namespace_reorder_spans.items())` looks unnecessary.